base/statistics.jl

```diff
@@ -388,18 +389,26 @@ cov(X::AbstractMatrix, Y::AbstractMatrix) = cov(X, Y, 1, true)
 
 ##### correlation #####
 
+"""
+    clampcor(x)
+
+Clamp a real correlation to between -1 and 1, leaving complex correlations unchanged
+"""
+clampcor(x::Real) = clamp(x, -1, 1)
+clampcor(x) = x
```

Comment:
It is possible that adjusting the number so that `abs(x) <= 1` decreases the accuracy of the real or imaginary components or the accuracy of `angle(x)`, and the `abs` operation itself may not even be correctly rounded; openlibm guarantees 1 ulp, not 0.5 ulp. So I think the appropriate adjustment would depend on what is relevant to the user (real component imaginary component, phase, and/or magnitude).

If one of the components is `>1` or `<-1` and the other is 0, then clamping that component would improve accuracy, but in any other case it's probably possible to find a counterexample. So we could do:

```julia
function clampcor(x::Complex)
    if imag(x) == 0 && abs(real(x)) > 1
        complex(copysign(one(real(x)), real(x)), zero(imag(x)))
    elseif real(x) == 0 && abs(imag(x)) > 1
        complex(zero(real(x)), copysign(one(imag(x)), imag(x)))
    else
        x
    end
end
```

This might make sense, since it would improve consistency between `cor(X)` and `cor(complex(X))`. But it might not be worth the computational cost, since these cases are probably relatively rare.